### PR TITLE
Implementation for "Modify Invoke-ReportExecution.ps1 to dump output in csv format"

### PR DESCRIPTION
**Pull request by prasadrao-dell**

### CODE REVIEW

File: Core/Python/update_firmware_using_catalog.py [hunk 1]
```diff
@@ -144,7 +154,40 @@ def catalog_creation_payload(**kwargs):
...
+
+def create_or_refresh_catalog(ip_address, headers, **kwargs):
+    """Get all catalogs first"""
+    url = 'https://%s/api/UpdateService/Catalogs' % ip_address
+    status, data = request(url=url, header=headers, method='GET')
+    if status != 200:
+        raise Exception("Unable to get the catalog", data)
+    else:
+        allrepoProfiles  = data["value"]
+        if allrepoProfiles is not None and data["@odata.count"] != 0:
```
Comment (RaajeevKalyan) [inline Core/Python/update_firmware_using_catalog.py:173]: I think "if allrepoProfiles" is sufficient - not if allrepoProfiles is not None

File: Core/Python/update_firmware_using_catalog.py [hunk 2]
```diff
@@ -144,7 +154,40 @@ def catalog_creation_payload(**kwargs):
...
+                    if kwargs['refresh']:
+                        url = 'https://%s/api/UpdateService/Actions/UpdateService.RefreshCatalogs' % ip_address
+                        refresh_payload = catalog_refresh_payload(repoProfile["Id"])
+                        status, data = request(url=url,
+                                               header=headers,  payload=refresh_payload,method='POST')
+                        if status != 204:
+                            raise Exception("Unable to refresh the catalog of " + repositoryType, data)
+                        else:
+                            time.sleep(60)
+                            print("Catalog refreshed ")
```
Comment (RaajeevKalyan) [inline Core/Python/update_firmware_using_catalog.py:186]: Is there a way to check if this completed / does this happen worst case within 60 seconds?

Comment (prasadrao-dell) [reply Core/Python/update_firmware_using_catalog.py:186]: The refresh job invocation appears to return a boolean and not a jobID.
But in several tests, the refresh completed within 60 seconds.

File: Core/Python/update_firmware_using_catalog.py (hunk 1 above)
Comment (prasadrao-dell) [reply Core/Python/update_firmware_using_catalog.py:173]: Done

**Comment by prasadrao-dell** [comment]
Review comments were discussed and addressed.

**Pull request merged**